### ISSUE TITLE
MD013: `reflow` with a non-default `reflow-mode` joins a setext h1 into its `===` underline, demoting it to a paragraph

### ISSUE THREAD

**Issue by schmidt-jake** [rvben/rumdl#825]
## Summary

With `MD013.reflow = true` and any explicitly-set non-default `reflow-mode`, `rumdl fmt` treats a setext h1 as a paragraph and joins the heading text onto its `===` underline. The heading is silently demoted to a paragraph — content loss of a structural element, with no warning.

## Version

`rumdl 0.2.58` (installed via `uv tool install rumdl`), macOS arm64. Also reproduced on Linux x86_64.

## Reproduction

`.rumdl.toml`:

```toml
[MD013]
reflow = true
reflow-mode = "normalize"
```

`repro.md`:

```markdown
Setup
=====

Some text.
```

Run:

```console
$ rumdl check repro.md
repro.md:1:1: [MD013] Paragraph could be normalized to use line length of 80 characters [*]

$ rumdl fmt repro.md
repro.md:1:1: [MD013] Paragraph could be normalized to use line length of 80 characters [fixed]
Fixed: Fixed 1/1 issues in 1 file
```

## Actual

```markdown
Setup =====

Some text.
```

Note the diagnostic itself says **"Paragraph"** — MD013 does not recognize the construct as a setext heading. That the heading is genuinely gone is confirmed by re-running `check` on the output:

```console
$ rumdl check repro.md
repro.md:1:1: [MD041] First line in file should be a level 1 heading
```

## Expected

The setext heading is left alone (or converted to ATX), and never merged with its underline.

## Scope

Tested against the same input on 0.2.58:

| config | result |
| --- | --- |
| `reflow = true` (mode unset) | ok |
| `reflow = true`, `reflow-mode = "default"` | ok |
| `reflow = true`, `reflow-mode = "normalize"` | **mangled** |
| `reflow = true`, `reflow-mode = "sentence-per-line"` | **mangled** |
| `reflow = true`, `reflow-mode = "semantic-line-breaks"` | **mangled** |
| `reflow = false`, `reflow-mode = "normalize"` | ok |

So it needs `reflow = true` **and** an explicitly-set non-`default` mode. `line-length` is not a factor (reproduces at the default 80 and at 100000).

Only `=`-underlined h1 is affected. The h2 form is fine:

```markdown
Subhead
-------
```

→ unchanged.

## Workaround

Setting `MD003.style = "atx"` masks it, since the heading is converted to `# Setup` before reflow sees it. That is easy to miss, though — a user who sets `reflow-mode` without also pinning MD003 loses headings silently, and `fmt` exits 0.

**Comment by rvben** [comment]
Fixed by 9ec9e174 and released in [v0.2.59](https://github.com/rvben/rumdl/releases/tag/v0.2.59). The release workflow completed successfully and the published artifacts are live.
